### ISSUE TITLE
Document how to use search and show with the global manifest

### ISSUE THREAD

**Issue by mkenigs** [flox/flox#958]
If there's an environment in the current directory or activated, there's no command that will allow searching or showing using the global manifest. Document this and suggest de-activating all environments and changing directory.

Also document that show operates in the context of an environment if there's one in the current dir or active

**Comment by mkenigs** [comment]
I don't know - I guess it depends on whether we expect users to actually understand the global base catalog, search, and show. My guess is that a first time user would think `search` just searches wherever `flox` gets packages. And then when they run into `flox update --global` or `flox search` giving them a prompt about which environment to use, they'll probably be confused until they build up some understanding. And then at that point they would expect these flags.

The global catalog has some other confusing behaviors like `flox init && flox update --global && flox install` is different than `flox init && flox install`. Not sure if we need to force people to learn how the global catalog works, or maybe we need to do a better job of hiding it

**Comment by ghudgins** [comment]
I see the premise of the global catalog and the environment specific catalog as a part of the product we don't try to emphasize but more in the 'how it works' part of the UX for the advanced cases where that information is important. so it's in the product but I really don't want most people to think about it

**Comment by mkenigs** [comment]
For the advanced case then, I think we'd either need to document how to get around environment detection (de-activate your environment or change directory), or we'd need to add and document these as advanced flags.

**Comment by ghudgins** [comment]
i'd be fine if we had the flags only in the man pages but i'm not sure if that breaks some convention. also would love to hear what @jennymahmoudi thinks

**Comment by jennymahmoudi** [comment]
Answering the question with a question... I understand the request, but I am not sure what the motivation is. I agree that the questions these flags might bring up would add more confusion than is worth for the convenience, based on my current understanding.

In the first case with `-r` and `-d`, is the concern that you are going to be prompted every time you try to search, and you think that will be annoying because you are going to be searching more than a couple times? Can we remember their choice, or ask to remember their choice for *this* scenario? Or am I misunderstanding the issue?

And for `global`... why would someone want to search globally? if we do that wouldn't we have to add more guardrails around which systems each package is available on? Or maybe we already have that problem?

**Comment by ghudgins** [comment]
>why would someone want to search globally?

you do this when you don't have an environment active or in your current directory--the global context is how search works and answers the question of which local revision is being searched (this will change with a service)

**Comment by jennymahmoudi** [comment]
the answer being... we will have a better solution when we have a catalog service?

**Comment by mkenigs** [comment]
> In the first case with `-r` and `-d`, is the concern that you are going to be prompted every time you try to search, and you think that will be annoying because you are going to be searching more than a couple times? Can we remember their choice, or ask to remember their choice for _this_ scenario? Or am I misunderstanding the issue?

Yeah there's currently no way to avoid the prompting. For any other environment command, you can add `-d` or `-r` to avoid the prompting. So for consistency, I would expect the same method of avoiding prompting to work here.

I think this would come up infrequently.

> And for `global`... why would someone want to search globally? if we do that wouldn't we have to add more guardrails around which systems each package is available on? Or maybe we already have that problem?

I think this would come up even more infrequently.

**Comment by ghudgins** [comment]
>the answer being... we will have a better solution when we have a catalog service?

yeah. the global context is at least a somewhat short lived thing ('now' will be the global context instead of some random state on your machine like it is now). searching in context of the environment is going to be a thing forever.

**Comment by jennymahmoudi** [comment]
> For the advanced case then, I think we'd either need to document how to get around environment detection (de-activate your environment or change directory), or we'd need to add and document these as advanced flags.

I would prefer we document the workarounds (without flags). That way, we are answering a question (how do I do this specific thing?), rather than creating a question (why would I want to do this specific thing?).

**Comment by zmitchell** [comment]
We no longer use a global manifest
